### PR TITLE
move items in a playlist

### LINKED ISSUE #197: shortcuts for moving items in a playlist?
first of all, epic work! really love it, it only takes 40-55mb ram, i have image & notify enabled, orig. spotify uses up to 800mb, sometimes even more ... crazy

i would wish me a feature, i'm change the order of my songs quite alot (at least at the beginning), would  it be possible to move the songs around?
something like:
"ctrl-arrowUp" & "ctrl-arrowDown" or with shift? or dwm-style with "j" & "k" or whatever makes sense to you ;)

thanks for all the work which went into this project, as soon as i get my pay, i'll sent you a coffee ;)

cheers
hardy

### PR DESCRIPTION
**Pull request by cobbinma**
add track actions to move items up and down in the current playlist 🎵

Screenshot 2023-07-26 at 18 18 30

closes #197

### CODE REVIEW

**Comment by aome510** [comment]
Hi @cobbinma, thanks for the PR. I think the original issue wants to implement the feature as commands instead of actions. I can find it quite troublesome, let's say, to move an item up 5 times as you need to call `GetAction`, wait for the popup then select the action. With command, you can just press a key.

It shouldn't be too difficult to change to command.

**Comment by cobbinma** [comment]
> Hi @cobbinma, thanks for the PR. I think the original issue wants to implement the feature as commands instead of actions. I can find it quite troublesome, let's say, to move an item up 5 times as you need to call `GetAction...` command, wait for the popup then select the action. With command, you can just press a key.
> 
> It shouldn't be too difficult to change to command.

Thanks @aome510 

I agree, that is much easier 👍

File: spotify_player/src/event/window.rs [hunk 1]
```diff
@@ -185,6 +185,66 @@ pub fn handle_command_for_track_table_window(
...
+            {
+                if id > 0
+                    && data
+                        .user_data
+                        .playlists
+                        .iter()
+                        .find(|playlist| &playlist.id == playlist_id)
+                        .is_some_and(|playlist| {
+                            Some(&playlist.owner.1)
+                                == data.user_data.user.as_ref().map(|user| &user.id)
```
Comment (aome510) [inline spotify_player/src/event/window.rs:202]: This owner check can be moved inside the `find` function right.

**Review by aome510 — CHANGES_REQUESTED** [review]
Looking good. Only need a minor change

**Comment by aome510** [comment]
I just tried the feature but the ordering does not persist after restart it changes ordering in the UI, which I think mainly because we also update the cached. @cobbinma does it work for you?

**Comment by cobbinma** [comment]
> I just tried the feature but the ordering change does not persist after restart. It does change the ordering in the UI though, which I think mainly because we also update the cached. @cobbinma does it work for you?

@aome510 It's only persisting moving upwards... 😕

**Comment by aome510** [comment]
> It's only persisting moving upwards... 😕

Oh right, also only persists moving upwards for me. Very interesting. Do you think you can investigate this behaviour? Otherwise, I think the PR is in a good state already.

**Comment by cobbinma** [comment]
> > It's only persisting moving upwards... 😕
> 
> Oh right, also only persists moving upwards for me. Very interesting. Do you think you can investigate this behaviour? Otherwise, I think the PR is in a good state already.

@aome510

I misunderstood the difference between insert before and index when using update playlist items.

Should be fixed now 👍

**Comment by aome510** [comment]
Thanks @cobbinma

**Pull request merged**